plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -1823,23 +1831,46 @@ public function delete( &$order, $args = array() ) {
 	}
 
 	/**
-	 * Helper method to set child orders to the parent order's parent.
+	 * Set the parent id of child orders to the parent order's parent if the post type
+	 * for the order is hierarchical, just delete the child orders otherwise.
 	 *
 	 * @param \WC_Abstract_Order $order Order object.
 	 *
 	 * @return void
 	 */
-	private function upshift_child_orders( $order ) {
+	private function upshift_or_delete_child_orders( $order ) {
 		global $wpdb;
-		$order_table  = self::get_orders_table_name();
-		$order_parent = $order->get_parent_id();
-		$wpdb->update(
-			$order_table,
-			array( 'parent_order_id' => $order_parent ),
-			array( 'parent_order_id' => $order->get_id() ),
-			array( '%d' ),
-			array( '%d' )
-		);
+
+		$order_table     = self::get_orders_table_name();
+		$order_parent_id = $order->get_parent_id();
+
+		if ( $this->legacy_proxy->call_function( 'is_post_type_hierarchical', $order->get_type() ) ) {
+			$wpdb->update(
+				$order_table,
+				array( 'parent_order_id' => $order_parent_id ),
+				array( 'parent_order_id' => $order->get_id() ),
+				array( '%d' ),
+				array( '%d' )
+			);
+		} else {
+			// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+			$child_order_ids = $wpdb->get_col(
+				$wpdb->prepare(
+					"SELECT id FROM $order_table WHERE parent_order_id=%d",
+					$order->get_id()
+				)
+			);
+
+			foreach ( $child_order_ids as $child_order_id ) {
+				// We can't use wc_get_order here because we would get a "Invalid parent ID" error
+				// (and we don't need to load the full order from the database anyway).
+				$child_order = new \WC_Order();
+				$child_order->set_verify_parent_id( false );
+				$child_order->set_id( $child_order_id );
+				$child_order->delete( true );
+			}
+			// phpcs:enable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
```

Comment:
This comment should be moved to like 1861

Reply:
I must have been doing something wrong before, but now I see that I can use `wc_get_order` for the child orders and I don't get the "Invalid parent ID" errors. So I have simplified the code accordingly (including removing the new `verify_parent_id` methods).